api4/role.go

```diff
@@ -85,6 +86,31 @@ func patchRole(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if !utils.IsLicensed() && patch.Permissions != nil {
+		allowedPermissions := []string{
+			model.PERMISSION_CREATE_TEAM.Id,
+			model.PERMISSION_MANAGE_WEBHOOKS.Id,
+			model.PERMISSION_MANAGE_SLASH_COMMANDS.Id,
+			model.PERMISSION_MANAGE_OAUTH.Id,
+			model.PERMISSION_MANAGE_SYSTEM_WIDE_OAUTH.Id,
+		}
```

Comment:
Just confirming, but the idea is that we already allow controlling of these in TE from configuration settings and this exposes that existing functionality in this endpoint by allowing the modification of roles instead of the 'permission-skipping configuration setting checks' and 'role updating based on configuration' we were doing previously?

Reply:
Yup. That's exactly right.